### PR TITLE
Add multi cluster set up to deployment section

### PR DESCRIPTION
**Pull request by SmritiSatyanV**

### CODE REVIEW

**Review by kumare3 — COMMENTED** [review]
Cc @evalsocket can you help adapt this doc for helm? Cc @katrogan

**Comment by evalsocket** [comment]
https://github.com/flyteorg/flyte/issues/2107

**Comment by kumare3** [comment]
Ok then let's merge this one and mention that an update for helm is
Coming soon

File: rsts/deployment/cluster_config/customizable_resources.rst [hunk 1]
```diff
@@ -5,8 +5,10 @@ Adding New Customizable Resources
 ##################################
 
 For background on customizable resources, see the :ref:`deployment-cluster-config-general`.
```
Comment (samhita-alla) [inline rsts/deployment/cluster_config/customizable_resources.rst:7]: Can we have this as a note after "Examples of such resources include execution clusters, task resource defaults, and more."?

File: rsts/deployment/cluster_config/customizable_resources.rst [hunk 2]
```diff
@@ -5,8 +5,10 @@ Adding New Customizable Resources
...
-As a quick refresher, custom resources allow you to manage configurations for specific combinations of user projects,
-domains and workflows that override default values. Examples of such resources include execution clusters, task resource
+As a quick refresher, custom resources allow you to manage configurations for :ref:`specific combinations <multicluster-setup>` of user projects,
```
Comment (samhita-alla) [inline rsts/deployment/cluster_config/customizable_resources.rst:8]: I'm not sure if the reference attached is right.

File: rsts/deployment/cluster_config/customizable_resources.rst [hunk 3]
```diff
@@ -15,11 +17,13 @@ Example
 
 Let's say you want to inject a default priority annotation for your workflows. Perhaps you start off with a model where everything has a default priority but soon you realize it makes sense that workflows in your production domain should take higher priority than those in your development domain.
 
-Now one of your user teams comes to you and wants to ensure that their critical workflows have even higher priority than other production workflows. Here's how you could go about building a customizable priority designation.
+Now, one of your users team require critical workflows to have a higher priority than other production workflows. Here's how you could go about building a customizable priority designation.
```
Comment (samhita-alla) [inline rsts/deployment/cluster_config/customizable_resources.rst:20]:
```suggestion
Now, one of your user teams requires critical workflows to have a higher priority than other production workflows. Here's how you could go about building a customizable priority designation.
```

File: rsts/deployment/cluster_config/customizable_resources.rst [hunk 4]
```diff
@@ -15,11 +17,13 @@ Example
...
 
 Flyte IDL
 ^^^^^^^^^
-You'll want to introduce a new :std:ref:`matchable resource <protos/docs/admin/admin:matchableresource>`, including a unique enum value and proto message definition. For example
+Introduce a new :std:ref:`matchable resource <protos/docs/admin/admin:matchableresource>`, that includes a unique enum value and proto message definition.
```
Comment (samhita-alla) [inline rsts/deployment/cluster_config/customizable_resources.rst:24]:
```suggestion
Introduce a new :std:ref:`matchable resource <protos/docs/admin/admin:matchableresource>` that includes a unique enum value and proto message definition. 
```

File: rsts/deployment/cluster_config/customizable_resources.rst [hunk 5]
```diff
@@ -46,7 +50,7 @@ See the changes in this `file <https://github.com/flyteorg/flyteidl/commit/b1767
 Flyte admin
```
Comment (samhita-alla) [inline rsts/deployment/cluster_config/customizable_resources.rst:50]:
```suggestion
Flyte Admin
```

File: rsts/deployment/cluster_config/customizable_resources.rst [hunk 6]
```diff
@@ -46,7 +50,7 @@ See the changes in this `file <https://github.com/flyteorg/flyteidl/commit/b1767
...
 ^^^^^^^^^^^
 
-Once your idl changes are released, update the logic of flyteadmin to `fetch <https://github.com/flyteorg/flyteadmin/commit/60b4c876ea105d4c79e3cad7d56fde6b9c208bcd#diff-510e72225172f518850fe582149ff320R122-R128>`__ your new matchable priority resource and use it when creating executions or wherever makes sense for your use case.
+Once your IDL changes are released, update the logic of flyteadmin to `fetch <https://github.com/flyteorg/flyteadmin/commit/60b4c876ea105d4c79e3cad7d56fde6b9c208bcd#diff-510e72225172f518850fe582149ff320R122-R128>`__ your new matchable priority resource and use it while creating executions or in relevant use cases.
```
Comment (samhita-alla) [inline rsts/deployment/cluster_config/customizable_resources.rst:53]:
```suggestion
Once your IDL changes are released, update the logic of Flyte Admin to `fetch <https://github.com/flyteorg/flyteadmin/commit/60b4c876ea105d4c79e3cad7d56fde6b9c208bcd#diff-510e72225172f518850fe582149ff320R122-R128>`__ your new matchable priority resource and use it while creating executions or in relevant use cases.
```

File: rsts/deployment/cluster_config/customizable_resources.rst [hunk 7]
```diff
@@ -73,6 +77,6 @@ For example:
 
 Flytekit
 ^^^^^^^^
-For convenience, add a flyte-cli wrapper for updating your new attribute. See `this PR <https://github.com/flyteorg/flytekit/pull/174>`__ for the full required set of changes.
+For convenience, add a flyte-cli wrapper for updating your new attribute. See `this PR <https://github.com/flyteorg/flytekit/pull/174>`__ for the entire set of changes required.
```
Comment (samhita-alla) [inline rsts/deployment/cluster_config/customizable_resources.rst:80]: @evalsocket, do we have to replace this section with FlyteCTL? Can you help with the content?

File: rsts/deployment/index.rst [hunk 8]
```diff
@@ -52,6 +52,15 @@ solution). The following pages will help you effectively deploy and manage an en
 
     ---
 
+    .. link-button:: multicluster
+       :type: ref
+       :text: Multi-cluster setup
+       :classes: btn-block stretched-link
+    ^^^^^^^^^^^^
+    Detailed instructions on multiple clusters setup.
```
Comment (samhita-alla) [inline rsts/deployment/index.rst:60]:
```suggestion
    Detailed instructions on multiple clusters' setup.
```
Is this correct? @SmritiSatyanV

**Comment by evalsocket** [comment]
@SmritiSatyanV I started working on helm changes, Let's publish docs with helm rather then kustomize https://github.com/flyteorg/flyte/pull/2169d

@kumare3 do we need docs for kustomize ?

File: rsts/deployment/cluster_config/customizable_resources.rst [hunk 9]
```diff
@@ -43,10 +46,10 @@ You'll want to introduce a new :std:ref:`matchable resource <protos/docs/admin/a
 See the changes in this `file <https://github.com/flyteorg/flyteidl/commit/b1767697705621a3fddcb332617a5304beba5bec#diff-d3c1945436aba8f7a76755d75d18e671>`__ for an example of what is required.
 
 
-Flyte admin
+FlyteAdmin
 ^^^^^^^^^^^
 
-Once your idl changes are released, update the logic of flyteadmin to `fetch <https://github.com/flyteorg/flyteadmin/commit/60b4c876ea105d4c79e3cad7d56fde6b9c208bcd#diff-510e72225172f518850fe582149ff320R122-R128>`__ your new matchable priority resource and use it when creating executions or wherever makes sense for your use case.
+Once your IDL changes are released, update the logic of Flyte Admin to `fetch <https://github.com/flyteorg/flyteadmin/commit/60b4c876ea105d4c79e3cad7d56fde6b9c208bcd#diff-510e72225172f518850fe582149ff320R122-R128>`__ your new matchable priority resource and use it while creating executions or in relevant use cases.
```
Comment (samhita-alla) [inline rsts/deployment/cluster_config/customizable_resources.rst:52]:
```suggestion
Once your IDL changes are released, update the logic of FlyteAdmin to `fetch <https://github.com/flyteorg/flyteadmin/commit/60b4c876ea105d4c79e3cad7d56fde6b9c208bcd#diff-510e72225172f518850fe582149ff320R122-R128>`__ your new matchable priority resource and use it while creating executions or in relevant use cases.
```

File: rsts/deployment/cluster_config/customizable_resources.rst [hunk 10]
```diff
@@ -73,6 +76,6 @@ For example:
 
 Flytekit
 ^^^^^^^^
-For convenience, add a flyte-cli wrapper for updating your new attribute. See `this PR <https://github.com/flyteorg/flytekit/pull/174>`__ for the full required set of changes.
+For convenience, add a FlyteCTL wrapper to update the new attributes. Refer to `this section <https://docs.flyte.org/projects/flytectl/en/latest/gen/flytectl_update_plugin-override.html>`__ for the entire set of changes required.
```
Comment (samhita-alla) [inline rsts/deployment/cluster_config/customizable_resources.rst:79]:
```suggestion
For convenience, add a FlyteCTL wrapper to update the new attributes. Refer to `this PR <https://github.com/flyteorg/flytectl/pull/65>`__ for the entire set of changes required.
```

File: rsts/deployment/cluster_config/customizable_resources.rst [hunk 11]
```diff
@@ -73,6 +76,6 @@ For example:
 
 Flytekit
 ^^^^^^^^
-For convenience, add a flyte-cli wrapper for updating your new attribute. See `this PR <https://github.com/flyteorg/flytekit/pull/174>`__ for the full required set of changes.
+For convenience, add a FlyteCTL wrapper to update the new attributes. Refer to `this PR <https://github.com/flyteorg/flytectl/pull/65>`__ for the entire set of changes required.
 
-That's it! You now have a new matchable attribute you can configure as the needs of your users evolve.
\ No newline at end of file
+That's it! You now have a new matchable attribute to configure according to the requirements of your users evolve.
```
Comment (samhita-alla) [inline rsts/deployment/cluster_config/customizable_resources.rst:81]:
```suggestion
That's it! You now have a new matchable attribute to configure as the needs of your users evolve.
```

File: rsts/deployment/cluster_config/customizable_resources.rst [hunk 12]
```diff
@@ -4,22 +4,25 @@
 Adding New Customizable Resources
 ##################################
 
-For background on customizable resources, see the :ref:`deployment-cluster-config-general`.
 As a quick refresher, custom resources allow you to manage configurations for specific combinations of user projects,
 domains and workflows that override default values. Examples of such resources include execution clusters, task resource
-defaults, and :std:ref:`more <flyteidl:protos/docs/admin/admin:matchableresource>`.
+defaults, and :std:ref:`more <flyteidl:protos/docs/admin/admin:matchableresource>`. If your goal is scalability, use :ref:`multiple clusters <multicluster-setup>`.
```
Comment (samhita-alla) [inline rsts/deployment/cluster_config/customizable_resources.rst:9]:
```suggestion
defaults, and :std:ref:`more <flyteidl:protos/docs/admin/admin:matchableresource>`.
```

**Review by samhita-alla — APPROVED** [review]

**Pull request merged**